engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -101,13 +104,13 @@ public T getValue() {
     }
 
     public boolean setValue(T newValue) {
-        if (!validator.validate(newValue))
+        if (!validator.validate(newValue)) {
+            LOGGER.warn("The passed value {} is invalid.", newValue);
```

Comment:
My apologies. In my earlier comment I did mean option 2, which is what you suggest. Specifically, I'd like:

```java
public final boolean validate(T value) {
    boolean isValid = fastValidate(value);
    if (!isValid) {
        this.issueWarnings(value);
    }
    return isValid;
}
```

where fastValidate and issueWarnings are methods that _must_ be implemented by subclasses.

Notice I could have suggested issueWarnings() - with no arguments- which would assume some preliminary works gets done in plainValidate, i.e. to select which message to send to the logger. 

In practice I would prefer for the fastValidate method to remain as fast possible. issueWarnings can then do whatever it needs, potentially replicating some of the checks done in fastValidate, to determine the exact message to log. In the most straightforward cases however, i.e. intervals, it might not involve any duplicate checks and be a one-liner looking like this:

```java
private void issueWarnings(T value) {
    // neglecting actual syntax for simplicity
    logger.warn( Cannot set module:myParam to 2.0: allowed interval [0.0, 2.0[ );
}
```